api-mock/fetcher.ts

```diff
@@ -153,56 +153,47 @@ export async function queryAPI({
     headers?: Record<string, string>;
     returnResponse?: boolean;
     skipInit?: boolean;
-}) {
+}): Promise<any> {
     if (!skipInit && apiStatus === "NOT_READY") {
         await initApp();
     }
-    try {
-        let response = await fetch(`http://localhost:${API_PORT}${path}`, {
-            method,
-            headers: {
-                "Content-Type": "application/json",
-                "fdi-version": fdiVersion,
-                ...headers,
-            },
-            body: JSON.stringify(input),
-        });
 
-        if (returnResponse) {
-            return response;
-        }
+    const response = await fetch(`http://localhost:${API_PORT}${path}`, {
+        method,
+        headers: {
+            "Content-Type": "application/json",
+            "fdi-version": fdiVersion,
+            ...headers,
+        },
+        body: JSON.stringify(input),
+    });
 
-        if (!response.ok) {
-            throw response;
-        }
+    // Return response as-is if requested
+    if (returnResponse) {
+        return response;
+    }
 
-        try {
-            const text =  await response.text();
-            // TODO: we need this for legacy tests (which should probably be updated)
-            if (text === "") {
-                return undefined;
-            }
-            
-            try {   
-                return JSON.parse(text);
-            } catch {
-                return text;
-            }
-        } catch {
-            return undefined;
-        }
-    } catch (error) {
-        if (error instanceof Response) {
-            const text = await error.text();
-            let errorBody;
-            try {
-                errorBody = JSON.parse(text);
-            } catch (e) {
-                throw new Error(text);
-            }
-            throw errorBody;
-        }
-        throw error;
+    // Clone response so we can re-use it below
+    const text = await response.clone().text();
+
+    // TODO: we need this for legacy tests (which should probably be updated)
+    if (text === "") {
+        return undefined;
+    }
+
+    if (!response.ok) {
+        // Response was not OK
+        // Parse the output as JSON and throw it as an error
+        // The parsing could error out itself, but we let that error bubble up
+        const errorBody = await response.json();
```

Comment:
this doesn't always work, we may want to do `throw new Error(text)` in those cases.

Reply:
Thanks, missed this
Added in 1374800